sim/battle.ts

```diff
@@ -1876,15 +1876,25 @@ export class Battle extends Dex.ModdedDex {
 
 		// In Gen 1 BUT NOT STADIUM, Substitute also takes confusion and HJK recoil damage
 		if (this.gen <= 1 && this.currentMod !== 'stadium' &&
-			['confusion', 'jumpkick', 'highjumpkick'].includes(effect.id) && target.volatiles['substitute']) {
-			target.volatiles['substitute'].hp -= damage;
-			if (target.volatiles['substitute'].hp <= 0) {
-				target.removeVolatile('substitute');
-				target.subFainted = true;
+			['confusion', 'jumpkick', 'highjumpkick'].includes(effect.id) &&
+			source && source.volatiles['substitute']) {
```

Comment:
I believe `target` is the user and `source` is the opponent in this scenario. Might also have to pass the opponent as `source` in the respective calls to directDamage to make it work.

Reply:
AKA "write a test next time, doofus" ;)

So yeah, after writing the test I think it works for confusion damage (though it looks like it maybe always worked for confusion damage, because it was implemented differently?), but not for (H)JK because `source === target` in `onMoveFail`. Any suggestions?